WooCommerce/Classes/Extensions/UIViewController+TableViewHelpers.swift

```diff
@@ -0,0 +1,22 @@
+import UIKit
+
+extension UIViewController {
+    /// Called in `viewDidLayoutSubviews`. If input table view has a footer view, calculates the new height.
+    /// If new height is different from current height, updates the footer view with the new height and reassigns the table footer view.
+    /// Note: make sure the top-level footer view (`tableView.tableFooterView`) is frame based as a container of the Auto Layout based subview.
+    ///
+    /// - Parameter tableView: a table view that is a subview of the view controller.
+    func updateFooterHeight(for tableView: UITableView) {
```

Comment:
How would you feel about making this an extension of UITableView instead of UIViewController? Somehow it seems to me like that would be more cohesive

Reply:
yep good question, it's not dependent on the view controller. will update!

Reply:
updated in https://github.com/woocommerce/woocommerce-ios/pull/1048/files/eb4707d17249f99fd6789eee7cca5041f4f423fd#diff-acdc002fddfcc830a3990d003accddb5